modmail/extensions/configuration_manager.py

```diff
@@ -0,0 +1,221 @@
+    async def config_group(self, ctx: Context) -> None:
+        """Manage the bot configuration."""
+        if ctx.invoked_subcommand is None:
+            await ctx.send_help(ctx.command)
+
+    @config_group.command(name="list")
+    async def list_config(self, ctx: Context) -> None:
+        """List the valid configuration options."""
+        options = {}
+        for table, opt in self.config_fields.items():
+            if opt.hidden or opt.frozen:
+                continue
+
+            # we want to merge items from the same config table so they are on the same table
+            key = table.rsplit(".", 1)[0]
+            options[key] = options.get(key, "") + (
+                "\n".join(
+                    [
+                        f"**{string.capwords(opt.canonical_name or opt.name)}**",
+                        f"Default: `{opt.default}`"
+                        if opt.default is not None
+                        else "Required. There is no default value for this option.",
+                        f"{opt.description}",
+                        f" {opt.extended_description}\n" if opt.extended_description else "",
+                    ]
+                )
+            )
+
+        await ButtonPaginator.paginate(options.values(), ctx.message)
+
+    @config_group.command(name="set", aliases=("edit",))
+    async def modify_config(self, ctx: Context, option: KeyConverter, value: str) -> None:
+        """Modify an existing configuration value."""
+        if option not in self.config_fields:
+            raise commands.BadArgument(f"Option must be in {', '.join(self.config_fields.keys())}")
+        meta = self.config_fields[option]
+
+        if meta.frozen:
+            # TODO: replace with responses module.
+            await ctx.send("Can't modify this value.")
+            return
+
+        if meta.modmail_metadata.discord_converter is not None:
+            value = await meta.modmail_metadata.discord_converter().convert(ctx, value)
+            if meta.modmail_metadata.discord_converter_attribute is not None:
+                if isinstance(meta.modmail_metadata.discord_converter_attribute, types.FunctionType):
+                    value = meta.modmail_metadata.discord_converter_attribute(value)
+        if meta._type in commands_converter.CONVERTER_MAPPING:
+            value = await commands_converter.CONVERTER_MAPPING[meta._type]().convert(ctx, value)
+            if isinstance(meta.modmail_metadata.discord_converter_attribute, types.FunctionType):
+                value = meta.modmail_metadata.discord_converter_attribute(value)
+        elif meta._field.converter:
+            value = meta._field.converter(value)
+
+        try:
+            root, name = option.rsplit(".", -1)
+        except ValueError:
+            root = ""
+            name = option
+
+        setattr(operator.attrgetter(root)(self.bot.config.user), name, value)
+
+        await ctx.message.reply("ok.")
```

Comment:
responses module again -- if we're waiting on a pr here, also mark this as TODO so it's not missed

Reply:
Added, good eye